# Práctico 3

En el área de la Astronomía existen diversas Bases de Datos como herramienta para almacenar todos los datos recolectados y poder luego acceder a ellos. Al manipular estos datos es necesario utilizar un software o lenguaje específico. En este caso se empleará el SQL (Structured Query Languaje) y la base de datos exoplanet.eu que es un catálogo de exoplanetas.

Para empezar, se creará una lista de exoplanetas que contenga las siguientes propiedades: masa, período orbital y tipo de estrella del sistema.

Primero se importará el paquete pyvo y se seleccionará el servicio a utilizar.

In [1]:
import pyvo

In [2]:
service = pyvo.dal.TAPService("http://voparis-tap-planeto.obspm.fr/tap")

A continuación se realizará un query o consulta a la base de datos exoplanet.eu para obtener las propiedades que se requieren.

In [3]:
query = "SELECT target_name, mass, period, star_spec_type FROM exoplanet.epn_core"
propiedades = service.search(query)

#se puede ver el resultado:
propiedades

<Table length=4315>
target_name       mass      period  star_spec_type
             'jupiterMass'    d                   
   object       float64    float64      object    
------------ ------------- -------- --------------
    11 Com b            --   326.03         G8 III
    11 Oph b          21.0 730000.0             M9
    11 UMi b            --   516.22          K4III
    14 And b            --   185.84          K0III
    14 Her b            --   1773.4           K0 V
  16 Cyg B b            --    799.5         G2.5 V
    17 Sco b            --   578.38         K3 III
    18 Del b            --    993.3          G6III
1I/'Oumuamua            --       --               
         ...           ...      ...            ...
   tau Cet e            --   162.87         G8.5 V
   tau Cet f            --   636.13         G8.5 V
   tau Cet g            --     20.0         G8.5 V
   tau Cet h            --    49.41         G8.5 V
   tau Gem b            --    305.5          K2III
   ups And 

Se puede notar que existen 4315 exoplanetas con estos datos.

In [6]:
#se puede ver qué características tiene "propiedades"
help(propiedades)

Help on TAPResults in module pyvo.dal.tap object:

class TAPResults(pyvo.dal.adhoc.DatalinkResultsMixin, pyvo.dal.query.DALResults)
 |  TAPResults(votable, url=None, session=None)
 |  
 |  The list of matching images resulting from an image (SIA) query.
 |  Each record contains a set of metadata that describes an available
 |  image matching the query constraints.  The number of records in
 |  the results is available via the :py:attr:`nrecs` attribute or by
 |  passing it to the Python built-in ``len()`` function.
 |  
 |  This class supports iterable semantics; thus,
 |  individual records (in the form of
 |  :py:class:`~pyvo.dal.query.Record` instances) are typically
 |  accessed by iterating over an ``TAPResults`` instance.
 |  
 |  Alternatively, records can be accessed randomly via
 |  :py:meth:`getrecord` or through a Python Database API (v2)
 |  Cursor (via :py:meth:`~pyvo.dal.query.DALResults.cursor`).
 |  Column-based data access is possible via the
 |  :py:meth:`~pyvo.dal.qu

A continuación se incluirán los datos en listas.

In [8]:
help(propiedades.getcolumn)

Help on method getcolumn in module pyvo.dal.query:

getcolumn(name) method of pyvo.dal.tap.TAPResults instance
    return a numpy array containing the values for the column with the
    given name



In [15]:
masa_prueba = propiedades.getcolumn(name = 'mass')
masa_prueba

masked_array(data=[--, 21.0, --, ..., 23.58, --, 40.0],
             mask=[ True, False,  True, ..., False,  True, False],
       fill_value=1e+20)

In [16]:
help(propiedades.getvalue)

Help on method getvalue in module pyvo.dal.query:

getvalue(name, index) method of pyvo.dal.tap.TAPResults instance
    return the value of a record attribute--a value from a column and row.
    
    Parameters
    ----------
    name : str
       the name of the attribute (column)
    index : int
       the zero-based index of the record
    
    Raises
    ------
    IndexError
       if index is negative or equal or larger than the
       number of rows in the result table.
    KeyError
       if name is not a recognized column name



In [20]:
i=0
masa = []
periodo = []
tipo_estrella = []

for i in range(len(propiedades)):
    masa.append(propiedades.getvalue(name = 'mass', index = i))
    periodo.append(propiedades.getvalue(name = 'period', index = i))
    tipo_estrella.append(propiedades.getvalue(name = 'star_spec_type', index = i))

In [21]:
# ver que hice 3 listas. Tengo que hacer una sola con todas las propiedades? una tabla con pandas por ejemplo?
#también corregir si hace falta sacar los exoplanetas que no tienen valores (nan)

# Tipo de detección

Se estudiará la distribución de índices de masas y períodos orbitales según las diferentes técnicas observacionales. De esta manera se podrá determinar si existe algún sesgo en cuanto a la técnica utilizada para la detección de exoplanetas al determinar sus propiedades.

Para esto es necesario consultar en la base de datos la técnica de detección utilizada por lo que se hará un nuevo query.

In [ ]:
# ver práctico borrador